nomad/fsm.go

```diff
@@ -2069,6 +2073,36 @@ func (n *nomadFSM) applyACLRolesDeleteByID(msgType structs.MessageType, buf []by
 	return nil
 }
 
+func (n *nomadFSM) applyACLAuthMethodsUpsert(buf []byte, index uint64) interface{} {
+	defer metrics.MeasureSince([]string{"nomad", "fsm", "apply_acl_auth_method_upsert"}, time.Now())
+	var req structs.ACLAuthMethodUpsertRequest
+	if err := structs.Decode(buf, &req); err != nil {
+		panic(fmt.Errorf("failed to decode request: %v", err))
+	}
+
+	if err := n.state.UpsertACLAuthMethods(index, req.AuthMethods); err != nil {
+		n.logger.Error("UpsertACLAuthMethods failed", "error", err)
+		return err
+	}
+
+	return nil
+}
+
+func (n *nomadFSM) applyACLAuthMethodsDelete(buf []byte, index uint64) interface{} {
```

Comment:
Right, so this makes me think: should we just simplify `DeleteACLAuthMethodsByName` to `DeleteACLAuthMethods` (same for respective `Get`)? since we don't delete or get by IDs or anything else?

Reply:
That's a good shout; mostly consistency across them will be good. Naming is hard!

Reply:
I simplified by dropping `ByName` from all method names and RPCs like we agreed.